### PR TITLE
Fixed CSVDataset and Added Tests

### PR DESCRIPTION
**Pull request by hassiahk**
As discussed, raised a new PR and this fixes https://github.com/Quick-AI/quickvision/issues/35

### CODE REVIEW

**Comment by oke-aditya** [comment]
Now the update branch button would make it super simple to update code.

**Comment by hassiahk** [comment]
Where is this update branch button?

I usually do the following to update the branch:
- `git pull upstream master`
- `git push origin branch-name`

This will not suffice?

P.S: I am a bit new to `git` :sweat_smile:. Forgive me for my unfamiliarity.

**Comment by oke-aditya** [comment]
Well, this just does that for you.

**Comment by hassiahk** [comment]
I now understood what you said. I just now need to do `git pull origin branch-name` locally because the changes were already reflected in the remote.

**Comment by oke-aditya** [comment]
Yep absolutely.

**Comment by oke-aditya** [comment]
The code looks fine to me, but I think we need to think with datasets API in general.
[Here](https://join.slack.com/t/quickai/shared_invite/zt-iz7tqk3r-IQa4SoxJGIK5WS8VdZhzeQ) is slack link where we can discuss

**Comment by hassiahk** [comment]
@oke-aditya, you can merge this for now if you are fine with the tests and we can build on top of it, the things we have discussed.

File: quickvision/datasets/classification.py [hunk 1]
```diff
@@ -40,26 +40,31 @@ class CSVDataset(Dataset):
     """
```
Comment (oke-aditya) [inline quickvision/datasets/classification.py:40]: Can this go under classification folder?

Comment (oke-aditya) [inline quickvision/datasets/classification.py:40]: Can you rename this with `CamelCase` as `CSVImageidDataset` ? or something similar ?

File: tests/test_datasets.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+from quickvision.datasets.classification import CSVDataset
+
+df = pd.read_csv('tests/assets/csv_dataset.csv')
+data_dir = 'tests/assets/'
+tfms = T.Compose([
+    T.ToTensor(),
+])
+
+
+class DatasetsTester(unittest.TestCase):
```
Comment (oke-aditya) [inline tests/test_datasets.py:13]: Maybe call it `ClassificationDatasetsTester`?  coz we would have similar for Detection etc.

File: tests/test_datasets.py [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    def test_csv_dataset(self):
+        flag = False
+        complete_dataset = CSVDataset(df, data_dir, 'Image', 'Label', tfms, 'png')
+        complete_dataset[0]
```
Comment (oke-aditya) [inline tests/test_datasets.py:17]: you can do `self.assertEquals(torch.utils.data.Dataset, complete_dataset)`

File: tests/test_datasets.py [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+        flag = True
+        self.assertTrue(flag)
```
Comment (oke-aditya) [inline tests/test_datasets.py:19]: Hmm, can you do a `sanity_fit` for CNN model by creating `Dataloader` ?
`sanity_fit` is very efficient, it does not calculate gradients and is fast to test over CPU.
Datasets have to be compatible with our models.

**Review by oke-aditya — CHANGES_REQUESTED** [review]

File: quickvision/datasets/classification.py (hunk 1 above)
Comment (hassiahk) [reply quickvision/datasets/classification.py:40]: Yes, it can since this is for single-label classification.

File: tests/test_datasets.py (hunk 3 above)
Comment (hassiahk) [reply tests/test_datasets.py:17]: It says this `AssertionError: <class 'torch.utils.data.dataset.Dataset'> != <quickvision.datasets.classification.CSVDataset object` when I do that.

File: quickvision/datasets/classification.py (hunk 1 above)
Comment (hassiahk) [reply quickvision/datasets/classification.py:40]: I am renaming it to `CSVSingleLabelDataset`.

File: tests/test_datasets.py (hunk 2 above)
Comment (hassiahk) [reply tests/test_datasets.py:13]: I will call it `CSVSingleLabelDatasetTester`

File: tests/test_datasets.py (hunk 4 above)
Comment (hassiahk) [reply tests/test_datasets.py:19]: I am doing `train_sanity_fit` instead of `sanity_fit`, since I only took two images to test this.

**Comment by hassiahk** [comment]
Made some changes, kindly review it once again @oke-aditya.

**Comment by oke-aditya** [comment]
It looks good. I think it's very well done.
I guess this is good start, for next datasets / items as we discussed. We should first have a look at a few possible and common scenarios which are possible.

These datasets are really useful additions and hence make CV tasks much easier and faster.

**Comment by hassiahk** [comment]
Yes, I am currently on it. I will consolidate all the possible ways that we can handle using `datasets` and create a new issue. We can discuss more regarding this after that.

**Pull request merged**